File: backend/src/main/java/com/bookbook/domain/rent/dto/response/RentSimpleResponseDto.java
package com.bookbook.domain.rent.dto.response;

import com.bookbook.domain.rent.entity.Rent;
import com.bookbook.domain.rent.entity.RentStatus;
import lombok.Builder;
import lombok.NonNull;

import java.time.LocalDateTime;

@Builder
public record RentSimpleResponseDto(
        @NonNull Integer id,
        @NonNull Long lenderUserId,
        @NonNull RentStatus status,
        @NonNull String bookCondition,
        @NonNull String bookTitle,
        @NonNull String author,
        @NonNull String publisher,
        @NonNull LocalDateTime createdDate,
        @NonNull LocalDateTime modifiedDate
){
    public static RentSimpleResponseDto from(Rent rent) {
        return RentSimpleResponseDto.builder()
                .id(rent.getId())
                .lenderUserId(rent.getLenderUserId())
                .status(rent.getRentStatus())
                .bookCondition(rent.getBookCondition())
                .bookTitle(rent.getBookTitle())
                .author(rent.getAuthor())
                .publisher(rent.getPublisher())
                .createdDate(rent.getCreatedDate())
                .modifiedDate(rent.getModifiedDate())
                .build();
    }
}

File: backend/src/main/java/com/bookbook/domain/rent/dto/response/RentSimpleResponseDto.kt
package com.bookbook.domain.rent.dto.response

import com.bookbook.domain.rent.entity.Rent
import com.bookbook.domain.rent.entity.RentStatus
import java.time.LocalDateTime

// 25.08.28 현준
// 대여 게시글의 간단한 정보를 담는 DTO.
data class RentSimpleResponseDto(
    val id: Long,
    val lenderUserId: Long,
    val status: RentStatus,
    val bookCondition: String,
    val bookTitle: String,
    val author: String,
    val publisher: String,
    val createdDate: LocalDateTime,
    val modifiedDate: LocalDateTime
) {
    constructor(rent: Rent) : this(
        id = rent.id,
        lenderUserId = rent.lenderUserId,
        status = rent.rentStatus,
        bookCondition = rent.bookCondition,
        bookTitle = rent.bookTitle,
        author = rent.author,
        publisher = rent.publisher,
        createdDate = rent.createdDate,
        modifiedDate = rent.modifiedDate
    )
}